react/src/pages/MainMenu/MainMenu.tsx

```diff
@@ -1,13 +1,36 @@
-import React from 'react';
+import React, { useState } from 'react';
 import {
   LoadingSpinner,
   InlineMessage,
   SectionHeader,
+  Button,
 } from '../../core-components';
 import { useProjects } from '../../hooks';
+import MapModal from '../../components/MapModal';
+import useCreateProject from '../../hooks/projects/useCreateProject';
 
 function MainMenu() {
   const { data, isLoading, error } = useProjects();
+  const { mutate: createProject, isLoading: isCreatingProject } =
+    useCreateProject();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const toggleModal = () => setIsModalOpen(!isModalOpen);
+
+  const handleCreateProject = (projectData) => {
+    createProject(projectData, {
+      onSuccess: () => {
+        // TO-ADD: Handle success (e.g., show success message, refetch projects, etc.)
```

Comment:
we want to navigate to the new map as now we know the new uuid:

```
import { useNavigate } from 'react-router-dom';
import { Project } from '../../types';

....

      onSuccess: (newProject: Project) => {
        navigate(`/project/${newProject.uuid}`);
      },
```

this /project/uuid route is already supported in `react/src/AppRouter.tsx` and bring us to a MapProject page (with fixtures for map data)